src/arg_parser.ts

```diff
@@ -200,20 +193,67 @@ export function setOptionsMapInitialValues(
     // For each option signature specified ...
     split.forEach((signature: string) => {
       // ... trim leading/trailing spaces that might be in the signature ...
-      signature = signature.trim();
+      const openBracketNdx = signature.indexOf("[");
 
-      // ... and check to see if it takes in a value
-      if (signature.includes("[value]")) {
-        // If it takes in a value, then remove the `[value]` portion ...
-        signature = signature.replace(/\s+\[value\]/, "").trim();
-        // ... and set the option object's `takes_value` flag to true
+      if (openBracketNdx === -1) {
+        signature = signature.trim();
+      } else {
         optionObject.takes_value = true;
+        const sig = signature.substring(0, openBracketNdx).trim();
```

Comment:
question here. what's the reason behind tokenizing the string? is it mainly for validating that the option is written properly so it can be parsed? simply trying to get context here so i can better understand the goal and if there may be another way than tokenizing